lib/portlayer/storage/volume.go

```diff
@@ -43,21 +67,35 @@ type Volume struct {
 
 	// Namespace in the storage layer to look up this volume.
 	SelfLink *url.URL
+
+	// Backing device
+	Device Disk
 }
 
-// VolumeStorer is an interface to create, remove, enumerate, and get Volumes.
-type VolumeStorer interface {
-	// Creates a volume on the given volume store, of the given size, with the given metadata.
-	VolumeCreate(ctx context.Context, ID string, store *url.URL, capacityMB uint64, info map[string][]byte) (*Volume, error)
+// NewVolume creates a Volume
+func NewVolume(store *url.URL, ID string, device Disk) (*Volume, error) {
+	storeName, err := util.VolumeStoreName(store)
+	if err != nil {
+		return nil, err
+	}
 
-	// Get an existing volume via it's ID.
-	VolumeGet(ctx context.Context, ID string) (*Volume, error)
+	selflink, err := util.VolumeURL(storeName, ID)
+	if err != nil {
+		return nil, err
+	}
 
-	// Destroys a volume
-	VolumeDestroy(ctx context.Context, ID string) error
+	// Set the label to the md5 of the ID
+	h := md5.New()
```

Comment:
If I remember correctly disk labels are capped at 15 characters (could be wrong) and md5sum is 16. Worth noting, but might not be worth doing anything about it at this point in time.